src/mcp/index.ts

```diff
@@ -221,8 +221,12 @@ export class FirebaseMcpServer {
     projectId = projectId || "";
 
     const accountEmail = await this.getAuthenticatedUser();
-    if (tool.mcp._meta?.requiresAuth && !accountEmail) {
-      return mcpAuthError();
+    if (tool.mcp._meta?.requiresAuth) {
+      try {
+        await requireAuth(await this.resolveOptions());
```

Comment:
How it is different then calling `getAuthenticatedUser()`?

```
  async getAuthenticatedUser(): Promise<string | null> {
    try {
      return await requireAuth(await this.resolveOptions());
    } catch (e) {
      return null;
    }
  }
```

Reply:
requireAuth can return null when you are authenticated via a service account, but it errors when the user is not authenticated. getAuthenticated catches those errors and returns null. Under the previous logic, this triggered the conditional that returned mcpAuthError()

Reply:
Good call - refactored this. Now, getAuthenticatedUser() returns:
email for firebase login
'Application default credentials when using ADC
null for gcloud auth application-default login

and we skip the redundant requireAuth call.